File: tests/TUnit.Templates.Tests/Snapshots/InstantiationTest.TUnit.AspNet._.verified/TUnit.AspNet/TUnit.AspNet/WebApplicationFactory.cs
using Microsoft.AspNetCore.Mvc.Testing;
using TUnit.Core.Interfaces;

namespace TUnit.AspNet;

public class WebApplicationFactory : WebApplicationFactory<Program>, IAsyncInitializer
{
    public Task InitializeAsync()
    {
        _ = Server;

        return Task.CompletedTask;
    }
}

File: tests/TUnit.Templates.Tests/Snapshots/InstantiationTestWithFSharp.TUnit.AspNet.FSharp._.verified/TUnit.AspNet.FSharp/TUnit.AspNet.FSharp/WebApplicationFactory.fs
namespace TUnit.AspNet.FSharp

open System.Threading.Tasks
open Microsoft.AspNetCore.Mvc.Testing
open TUnit.Core.Interfaces

type WebApplicationFactory() =
    inherit WebApplicationFactory<WebApp.Program.Program>()

    interface IAsyncInitializer with
        member this.InitializeAsync() : Task =
            let _ = this.Server
            Task.CompletedTask


